src/Bicep.Core/Syntax/ImportSpecification.cs

```diff
@@ -46,35 +48,57 @@ private ImportSpecification(string name, string version, bool isValid, TextSpan
 
         public static ImportSpecification From(SyntaxBase specificationSyntax)
         {
-            switch (specificationSyntax)
+            return specificationSyntax switch
             {
-                case StringSyntax stringSyntax when stringSyntax.TryGetLiteralValue() is { } value:
-                    var (name, version, isValid) = Parse(value);
-                    var span = isValid ? new TextSpan(stringSyntax.Span.Position + 1, name.Length) : stringSyntax.Span;
-
-                    return new ImportSpecification(name, version, isValid, span);
-
-                case SkippedTriviaSyntax trivia:
-                    return new ImportSpecification(trivia.TriviaName, trivia.TriviaName, false, trivia.Span);
+                StringSyntax stringSyntax when stringSyntax.TryGetLiteralValue() is { } value
+                    => CreateFromStringSyntax(stringSyntax, value),
+                SkippedTriviaSyntax trivia
+                    => new ImportSpecification(
+                        trivia.TriviaName,
+                        trivia.TriviaName,
+                        trivia.TriviaName,
+                        false,
+                        trivia.Span),
+                _
+                    => new ImportSpecification(
+                        LanguageConstants.ErrorName,
+                        LanguageConstants.ErrorName,
+                        LanguageConstants.ErrorName,
+                        false,
+                        specificationSyntax.Span)
```

Comment:
Can we merge the `SkippedTriviaSyntax` case and the default case? Both of them indicate error and I'm not sure if it's worth distinguish them.